### PR TITLE
PLAT-4890 - Create a lib to authentication.

### PR DESCRIPTION
**Pull request by crepache**

### CODE REVIEW

File: symphony-app-authentication-fe/sagas/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+const baseAuthenticationUrl = "";
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/sagas/apiCalls.js]: I believe it's not required.

File: symphony-app-authentication-fe/sagas/apiCalls.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export default class ApiCalls {
+  constructor(baseAuthenticationUrl) {
+    this.baseAuthenticationUrl = `${baseAuthenticationUrl}/v1/application/jwt`
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/sagas/apiCalls.js]: this.baseAuthenticationUrl = `${baseAuthenticationUrl}/v1/application`

File: symphony-app-authentication-fe/sagas/apiCalls.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  authenticateApp(podId, appId) {
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/sagas/apiCalls.js]: Only appId will be required.

File: symphony-app-authentication-fe/sagas/apiCalls.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        applicationToken,
+        symphonyToken,
+        appId
+    };
+
+    return axios.post(url, payload);
+  }
+
+  validateJwt(jwt) {
+    const url = `${baseAuthenticationUrl}/validate`;
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/sagas/apiCalls.js]: const url = `${baseAuthenticationUrl}/jwt/validate`;

File: symphony-app-authentication-fe/services/bootstrapService.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+export const initApp = (config) => {
+  let userInfo = {};
+  let tokenA = '';
+
+  // create our own service
+  SYMPHONY.services.register(`${config.appId}:controller`);
+
+  
+  const authenticateApplication = (podInfo) => {
+    return authenticateApp(podInfo.pod, config.appId);
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/bootstrapService.js]: return authenticateApp(config.appId);

File: symphony-app-authentication-fe/services/bootstrapService.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  const validateJwtToken = (jwt) => {
+    userInfo.jwt = jwt;
+    
+    return validateJwt(jwt);
+  }
+
+  SYMPHONY.remote.hello()
+  .then(authenticateApplication)
+  .then(validateAppTokens)
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/bootstrapService.js]: You missed to register application before validate tokens.

File: symphony-app-authentication-fe/services/bootstrapService.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  .then(validateJwtToken)
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/bootstrapService.js]: You missed to cache the generated JWT locally. You should also provide an easy way to get this token once library consume will require this to perform API calls.

File: symphony-app-authentication-fe/services/registerApplication.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { Utils } from '../js/utils.service';
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/registerApplication.js]: This import is not required.

File: symphony-app-authentication-fe/services/registerApplication.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+const dependencies = [
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/registerApplication.js]: Harcoded? The library consumer should have the ability to provide a custom list of dependencies.  Only 'extended-user-info' service is required to retrieve JWT.

File: symphony-app-authentication-fe/webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
+/* eslint-disable */
+var webpack = require('webpack');
+var path = require("path"),
+  CopyWebpackPlugin = require('copy-webpack-plugin');
+
+module.exports = {
+  entry: {
+    // babelPolyfill: 'babel-polyfill',
+    controller: path.resolve(__dirname, "./src/main/webapp/js/controller.js")
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/webpack.config.js]: This file does not exists.

File: symphony-app-authentication-fe/webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  output: {
+    path: path.resolve(__dirname, "./target/static"),
+    filename: "[name].bundle.js"
+  },
+  devtool: 'source-map',
+  module: {
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/webpack.config.js]: Please review this config file. There are a lot of stuff not required like react presets, handlebars, css, images, fonts, etc.

File: symphony-app-authentication-fe/services/registerApplication.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+* @params       config                      app settings
+* @return       SYMPHONY.remote.hello       returns a SYMPHONY remote hello service.
+*/
+export const registerApplication = (config, appData) => {
+  const controllerName = `${config.appId}:controller`;
+
+  return SYMPHONY.application.register(
+          appData, 
+          dependencies, 
+        ).then(register);
```
Comment (robson-sanchez) [inline symphony-app-authentication-fe/services/registerApplication.js]: There is no 'register' function here. The library consumer should have the ability to provide a custom callback function to execute after this call or you should return this promise  after finishing the whole process.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: symphony-app-authentication-fe/sagas/apiCalls.js (hunk 3 above)
Comment (crepache) [reply symphony-app-authentication-fe/sagas/apiCalls.js]: Done.

File: symphony-app-authentication-fe/sagas/apiCalls.js (hunk 4 above)
Comment (crepache) [reply symphony-app-authentication-fe/sagas/apiCalls.js]: Done.

File: symphony-app-authentication-fe/services/bootstrapService.js (hunk 5 above)
Comment (crepache) [reply symphony-app-authentication-fe/services/bootstrapService.js]: Done.

File: symphony-app-authentication-fe/services/registerApplication.js (hunk 8 above)
Comment (crepache) [reply symphony-app-authentication-fe/services/registerApplication.js]: Done.

File: symphony-app-authentication-fe/sagas/apiCalls.js (hunk 2 above)
Comment (crepache) [reply symphony-app-authentication-fe/sagas/apiCalls.js]: Done.

File: symphony-app-authentication-fe/sagas/apiCalls.js (hunk 1 above)
Comment (crepache) [reply symphony-app-authentication-fe/sagas/apiCalls.js]: Done.

File: symphony-app-authentication-fe/services/bootstrapService.js (hunk 6 above)
Comment (crepache) [reply symphony-app-authentication-fe/services/bootstrapService.js]: Done.

File: symphony-app-authentication-fe/services/registerApplication.js (hunk 9 above)
Comment (crepache) [reply symphony-app-authentication-fe/services/registerApplication.js]: Done.

File: symphony-app-authentication-fe/webpack.config.js (hunk 10 above)
Comment (crepache) [reply symphony-app-authentication-fe/webpack.config.js]: Done.

File: symphony-app-authentication-fe/webpack.config.js (hunk 11 above)
Comment (crepache) [reply symphony-app-authentication-fe/webpack.config.js]: Done.

File: symphony-app-authentication-fe/services/registerApplication.js (hunk 12 above)
Comment (crepache) [reply symphony-app-authentication-fe/services/registerApplication.js]: Done.

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**